# KNN Classification

|Height|Weight|Gender|
|------|------|------|
|170|65|Male|
|175|70|Male|
|160|50|Female|
|155|45|Female|

In [ ]:
# New Person , Hegiht = 168, Weight = 62

- Step 1 : Choose K (K=3)
- Step 2 : Calculate Distance between new point and all training data points
- Step 3 : Select Nearest Neighbors
- Step 4 : Majority Voting (Male=2,Female=1)
- Step 5 : Prediction : Male

# KNN Regression

|House Size|Price|
|--|--|
|1000|500000|
|1200|600000|
|1300|650000|
|1500|750000|

In [3]:
# New House : Size = 1250, Predict Price
175/3

58.333333333333336

- Choose K (3)
- Find 3 nearest houses (1000,1200,1300)
- Average Output (175/3=58.33 lakhs)

Distance Metrics
- Euclidean Distance - straight line distance(airplane route, bird flight path)
- Manhattan Distance - movements happen horizontally and vertically(road networks, taxi routes, bike routes)